### PR TITLE
feat: add amino signing support to tokenfactory messages

### PR DESCRIPTION
**Pull request by madrezaz**
## What is the purpose of the change

This PR changes the codec being used for getting sign bytes of tokenfactory module messages, in order to allow amino signing.

## Testing and Verifying

This change is a trivial rework / code cleanup without any test coverage.

## Documentation and Release Note

  - [x] Does this pull request introduce a new feature or user-facing behavior changes?
  - [x] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

File: x/tokenfactory/types/codec.go [hunk 1]
```diff
@@ -25,24 +30,28 @@ func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
...
 	sdk.RegisterLegacyAminoCodec(amino)
-	RegisterCodec(authzcodec.Amino)
-
 	amino.Seal()
+
+	// Register all Amino interfaces and concrete types on the authz and gov Amino codec so that this can later be
+	// used to properly serialize MsgGrant, MsgExec and MsgSubmitProposal instances
+	RegisterLegacyAminoCodec(authzcodec.Amino)
+	RegisterLegacyAminoCodec(govcodec.Amino)
+	RegisterLegacyAminoCodec(groupcodec.Amino)
```
Comment (mattverse) [inline x/tokenfactory/types/codec.go:56]: annoying that we need to do this :(

**Review by mattverse — APPROVED** [review]
uTACK, thanks so much !

File: x/tokenfactory/types/codec.go (hunk 1 above)
Comment (madrezaz) [reply x/tokenfactory/types/codec.go:56]: yeah, I hope we can remove these in newer versions with textual sign mode

**Comment by mattverse** [comment]
> Considering you tagged the PR as state breaking, should I move the changelog entry to another section?

Should be fine :) thanks!

**Pull request merged**